pallets/moonbeam-orbiters/src/lib.rs

```diff
@@ -0,0 +1,500 @@
+						orbiter,
+					});
+				}
+				RemoveOrbiterResult::OrbiterRemoveScheduled => (),
+			}
+
+			CollatorsPool::<T>::insert(collator, collator_pool);
+			Ok(())
+		}
+
+		/// Registering as an orbiter
+		#[pallet::weight(500_000_000)]
+		pub fn orbiter_register(origin: OriginFor<T>) -> DispatchResult {
+			let orbiter = ensure_signed(origin)?;
+
+			if let Some(min_orbiter_deposit) = MinOrbiterDeposit::<T>::get() {
+				// The use of `ensure_reserved_named` allows to update the deposit amount in case a
+				// deposit has already been made.
+				T::Currency::ensure_reserved_named(
+					&T::OrbiterReserveIdentifier::get(),
+					&orbiter,
+					min_orbiter_deposit,
+				)
+			} else {
+				Err(Error::<T>::MinOrbiterDepositNotSet.into())
+			}
+		}
+
+		/// Deregistering from orbiters
+		#[pallet::weight(500_000_000)]
+		pub fn orbiter_unregister(
+			origin: OriginFor<T>,
+			collators_pool_count: u32,
+		) -> DispatchResult {
+			let orbiter = ensure_signed(origin)?;
+
+			// We have to make sure that the `collators_pool_count` parameter is large enough,
+			// because its value is used to calculate the weight of this extrinsic
+			ensure!(
+				collators_pool_count >= CollatorsPool::<T>::count(),
+				Error::<T>::CollatorsPoolCountTooLow
+			);
+
+			// Ensure that the orbiter is not in any pool
+			ensure!(
+				!CollatorsPool::<T>::iter_values()
+					.any(|collator_pool| collator_pool.contains_orbiter(&orbiter)),
+				Error::<T>::OrbiterStillInAPool,
+			);
+
+			T::Currency::unreserve_all_named(&T::OrbiterReserveIdentifier::get(), &orbiter);
+
+			Ok(())
+		}
+
+		/// Add a collator to orbiters program.
+		#[pallet::weight(500_000_000)]
+		pub fn add_collator(
+			origin: OriginFor<T>,
+			collator: <T::Lookup as StaticLookup>::Source,
+		) -> DispatchResult {
+			T::AddCollatorOrigin::ensure_origin(origin)?;
+			let collator = T::Lookup::lookup(collator)?;
+
+			ensure!(
+				CollatorsPool::<T>::get(&collator).is_none(),
+				Error::<T>::CollatorAlreadyAdded
+			);
+
+			CollatorsPool::<T>::insert(collator, CollatorPoolInfo::default());
+
+			Ok(())
+		}
+
+		/// Remove a collator from orbiters program.
+		#[pallet::weight(500_000_000)]
+		pub fn remove_collator(
+			origin: OriginFor<T>,
+			collator: <T::Lookup as StaticLookup>::Source,
+		) -> DispatchResult {
+			T::DelCollatorOrigin::ensure_origin(origin)?;
+			let collator = T::Lookup::lookup(collator)?;
+
+			// Remove the pool associated to this collator
+			let collator_pool =
+				CollatorsPool::<T>::take(&collator).ok_or(Error::<T>::CollatorNotFound)?;
+
+			// Remove all AccountLookupOverride entries related to this collator
+			for orbiter in collator_pool.get_orbiters() {
+				AccountLookupOverride::<T>::remove(&orbiter);
+			}
+			AccountLookupOverride::<T>::remove(&collator);
+
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		/// Notify this pallet that a new round begin
+		pub fn on_new_round(round_index: T::RoundIndex) -> Weight {
```

Comment:
I would split it:
`on_new_round`
and the core logic in
`on_rotate`

Reply:
Done